src/interface/drawstatus.cpp

```diff
@@ -2008,66 +2008,36 @@ void drawStatus(int player)
 
 		bool bumper_moved = false;
 
-		//Gamepad change hotbar selection.
-		if ( inputs.bControllerInputPressed(player, INJOY_GAME_HOTBAR_NEXT)
-			|| *inputPressedForPlayer(player, impulses[IN_HOTBAR_SCROLL_RIGHT]) )
+		Input& input = Input::inputs[player];
+
+		if ( input.consumeBinaryToggle("Hotbar Scroll Right") )
 		{
 			if ( shootmode && !inputs.getUIInteraction(player)->itemMenuOpen && !openedChest[player]
 				&& gui_mode != (GUI_MODE_SHOP) && !players[player]->bookGUI.bBookOpen
 				&& !GenericGUI[player].isGUIOpen() )
 			{
-				if ( *inputPressedForPlayer(player, impulses[IN_HOTBAR_SCROLL_RIGHT]) )
-				{
-					*inputPressedForPlayer(player, impulses[IN_HOTBAR_SCROLL_RIGHT]) = 0;
-				}
-				else
-				{
-					inputs.controllerClearInput(player, INJOY_GAME_HOTBAR_NEXT);
-					bumper_moved = true;
-				}
+				bumper_moved = true;
 				hotbar_t.hotbarTooltipLastGameTick = ticks;
 				players[player]->hotbar.selectHotbarSlot(players[player]->hotbar.current_hotbar + 1);
 			}
 			else
 			{
 				hotbar_t.hotbarTooltipLastGameTick = 0;
-				/*if ( intro || shootmode )
-				{
-					if ( *inputPressedForPlayer(player, impulses[IN_HOTBAR_SCROLL_RIGHT]) )
-					{
-						*inputPressedForPlayer(player, impulses[IN_HOTBAR_SCROLL_RIGHT]) = 0;
-					}
-				}*/
 			}
 		}
-		if ( inputs.bControllerInputPressed(player, INJOY_GAME_HOTBAR_PREV) || *inputPressedForPlayer(player, impulses[IN_HOTBAR_SCROLL_LEFT]) )
+		if ( input.consumeBinaryToggle("Hotbar Scroll Left") )
```

Comment:
bumper_moved is for warping the game controller's virtual mouse - so originally it was used to detect if a game controller button was pressed.

Confirm this doesn't cause the real mouse cursor to bounce around when you scroll the hotbar? I don't think it should but please confirm.

Reply:
Actually nevermind this is the old drawStatus() module - I will delete this function eventually